### PR TITLE
Issue 195: Use `brms::bf`

### LINKED ISSUE #195: Use formula in `brms` format i.e. `brms::bf` as input to `epidist`
> And should this instead be in the `brms` format? i.e. `brms::bf` or something like that. Rather than a list of `formula`

_Originally posted by @athowes in https://github.com/epinowcast/epidist/pull/183#discussion_r1689793381_

We should try to use the same input methods for formula as `brms` does. This would also mean that users wouldn't have to input formula for all variables, and it would know that if a formula is not provided then to use `not_provided ~ 1`.

### PR DESCRIPTION
**Pull request by athowes**
## Description

This draft PR will close #195.

## Checklist

- [ ] My PR is based on a package issue and I have explicitly linked it.
- [ ] I have included the target issue or issues in the PR title in the for Issue(s) *issue-numbers*: PR title
- [ ] I have read the [contribution guidelines](https://github.com/epinowcast/.github/blob/main/CONTRIBUTING.md).
- [ ] I have tested my changes locally.
- [ ] I have added or updated unit tests where necessary.
- [ ] I have updated the documentation if required.
- [ ] My code follows the established coding standards.
- [ ] I have added a news item linked to this PR.
- [ ] I have reviewed CI checks for this PR and addressed them as far as I am able.

### CODE REVIEW

**Review by seabbs — COMMENTED** [review]
This is very cool

File: R/latent_individual.R [hunk 1]
```diff
@@ -146,61 +146,24 @@ epidist_family.epidist_latent_individual <- function(data,
...
-  }
-  mu_index <- which(dpars == "mu")
-  formula[[mu_index]] <- stats::update(
-    formula[[mu_index]],
-    delay_central | vreal(obs_t, pwindow_upr, swindow_upr) ~ .
+  formula <- brms:::validate_formula(formula, data = data, family = family)
+  
+  # Using this here only for checking purposes: expect it catches some errors
+  bterms <- brms::brmsterms(formula)
+  brms:::validate_data(data, bterms)
```
Comment (athowes) [inline R/latent_individual.R:162]: Try moving this after

File: R/defaults.R [hunk 2]
```diff
@@ -75,7 +75,7 @@ epidist_stancode.default <- function(data, ...) {
 #' @method epidist default
 #' @family defaults
 #' @export
-epidist.default <- function(data, formula = list(mu ~ 1, sigma ~ 1),
+epidist.default <- function(data, formula = brms::bf(mu ~ 1, sigma ~ 1),
```
Comment (seabbs) [inline R/defaults.R:78]: can we entirely avoid specifying sigma and thus make this portable across all families? I think it should build down to the same thing?

Comment (athowes) [reply R/defaults.R:78]: Yes good point. Not sure. I'll try it and report back

Comment (athowes) [reply R/defaults.R:78]: No it doesn't work (breaks 7 tests)

Comment (athowes) [reply R/defaults.R:78]: I think this is solvable so will make an issue for it

Comment (athowes) [reply R/defaults.R:78]: https://github.com/epinowcast/epidist/issues/222

**Review by seabbs — APPROVED** [review]

**Pull request merged**